### PR TITLE
updated 1.2 version for alpine

### PR DESCRIPTION
**Pull request by totemofwolf**
* updated 1.2 version for alpine.
* add alpine build hooks
* fix issue: the alpine image is too fat #43

### CODE REVIEW

File: alpine/Dockerfile [hunk 1]
```diff
@@ -1,21 +1,39 @@
- FROM openresty/openresty:alpine-fat
+FROM node:10-alpine3.11 AS build-stage
```
Comment (moonming) [inline alpine/Dockerfile:1]: I think `openresty:alpine-fat` should be the build iamge, then copy OpenResty and APISIX files to `alpine:latest` image:

```
FROM openresty/openresty:alpine-fat
# install APISIX
FROM alpine:latest  
COPY OpenResty and APSIX dir
```

File: alpine/Dockerfile [hunk 2]
```diff
@@ -1,24 +1,57 @@
...
+
+ARG APISIX_VERSION=1.1
```
Comment (moonming) [inline alpine/Dockerfile:3]: the version should be 1.2

File: alpine/Dockerfile [hunk 3]
```diff
@@ -1,24 +1,57 @@
...
+    && git clone -b v${APISIX_VERSION} https://github.com/apache/incubator-apisix.git /tmp/apisix \
+    && cd /tmp/apisix \
+    && git submodule init \
+    && git submodule update \
+    && cd dashboard \
+    && yarn && yarn build:prod
+
+FROM openresty/openresty:alpine-fat AS production-stage
 
 ARG APISIX_VERSION=1.1
```
Comment (moonming) [inline alpine/Dockerfile:17]: the 1.1 version not has dashbaord. So I think build will break.

File: alpine/Dockerfile [hunk 4]
```diff
@@ -1,24 +1,57 @@
...
+    && bin='#! /usr/local/openresty/luajit/bin/luajit\npackage.path = "/usr/local/apisix/?.lua;" .. package.path' \
     && sed -i "1s@.*@$bin@" /usr/bin/apisix \
-    && mv /usr/local/apisix/deps/share/lua/5.1/apisix/lua /usr/local/apisix \
-    && apk del .builddeps
+    && mv /usr/local/apisix/deps/share/lua/5.1/apisix /usr/local/apisix \
+    && apk del .builddeps build-base make unzip
+
+FROM alpine:3.11 AS last-stage
+
+# add runtime for Apache Apisix
```
Comment (moonming) [inline alpine/Dockerfile:38]: `Apisix` -> `APISIX`

File: alpine/Dockerfile (hunk 2 above)
Comment (totemofwolf) [reply alpine/Dockerfile:3]: The repo file `incubator-apisix-docker/alpine/hooks/build` will take over the real build process and accept the variable `APISIX_VERSION` which set in dockerhub UI.

**Pull request merged**